File: backend/app/models/dao_document.py
from sqlalchemy import Column, Integer, Text, String, ForeignKey, DateTime, func
from sqlalchemy.orm import relationship

from ..database import Base


class DaoDocument(Base):
    __tablename__ = "dao_documents"

    id = Column(Integer, primary_key=True, index=True)
    tender_call_id = Column(Integer, ForeignKey("tender_calls.id"), nullable=False, unique=True)
    cahier_des_charges = Column(Text, nullable=True)
    criteres_selection = Column(Text, nullable=True)
    conditions_participation = Column(Text, nullable=True)
    pieces_exigees = Column(Text, nullable=True)
    document_url = Column(String(512), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    tender_call = relationship("TenderCall", back_populates="dossier_dao")

File: frontend/src/types/daoDocument.ts
import type { OfferDocumentType } from "./offerDocument";

export interface DaoDocument {
  id: number;
  tender_call_id: number;
  cahier_des_charges?: string | null;
  criteres_selection?: string | null;
  conditions_participation?: string | null;
  pieces_exigees?: string | null;
  required_document_types: OfferDocumentType[];
  document_url?: string | null;
  created_at: string;
  updated_at: string;
}

export interface DaoDocumentCreate {
  tender_call_id: number;
  cahier_des_charges?: string;
  criteres_selection?: string;
  conditions_participation?: string;
  pieces_exigees?: string;
  required_document_types?: OfferDocumentType[];
  document_url?: string;
}

export type DaoDocumentUpdate = Partial<Omit<DaoDocumentCreate, "tender_call_id">>;
